src/Symfony/Component/Cache/Adapter/Helper/FilesCacheHelper.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Adapter\Helper;
+
+use Symfony\Component\Cache\Exception\InvalidArgumentException;
+
+class FilesCacheHelper
```

Comment:
I had to move some logic away (I couldn't just delete files for hhvm), and I just thought it's going to be more readable this way. Plus you could add tests to it this way if it will grow larger.

I can move it back to abstract if you prefer, I don't mind.